### ISSUE TITLE
Training scripts use step-based LR scheduler horizons with interval='epoch'

### ISSUE THREAD

**Issue by mvpeterson** [galilai-group/stable-worldmodel#280]
There appears to be a learning-rate scheduler unit mismatch in [training scripts](https://github.com/galilai-group/stable-worldmodel/blob/b298aa704435f694669f498ec6df479ae853618f/scripts/train/lewm.py#L150).

These scripts compute scheduler horizons in optimizer-step units:

```
total_steps = cfg.trainer.max_epochs * len(train)
    optimizers = {
        'model_opt': {
            'modules': 'model',
            'optimizer': dict(cfg.optimizer),
            'scheduler': {
                'type': 'LinearWarmupCosineAnnealingLR',
                'warmup_steps': max(1, int(0.01 * total_steps)),
                'max_steps': total_steps,
            },
            'interval': 'epoch',
        },
    }
```
`total_steps` appears to mean: `num_epochs * num_batches_per_epoch`
So `warmup_steps` and `max_steps` are batch/optimizer-step counts.
But `'interval': 'epoch'` means the scheduler is likely stepped once per epoch.

So if you have 100 epochs and 1,000 batches per epoch: `total_steps=100_000 warmup_steps=1_000`
But the scheduler only gets called 100 times. It may stay in warmup the entire training run or behave very differently than intended.

Is 'interval': 'epoch' here intentional? Should interval be 'step' instead of 'epoch'?

If there is custom logic in the training wrapper that changes the meaning here, a clarification would be very helpful.

**Comment by lucas-maes** [comment]
Hi @mvpeterson! thank you very much for pointing out! would you able able to dig the code/ try to benchmark perf against the two approach in order to be sure?

**Comment by mvpeterson** [comment]
I ran a set of training sessions (`uv run python scripts/train/lewm.py`) using the [galilai-group/lewm-pusht](https://huggingface.co/datasets/galilai-group/lewm-pusht) dataset to compare epoch-based and step-based scheduling.

It didn't show a significant difference in validation statistics, and the LR curves are identical. Possibly, your config wrapper is handling the `'interval': 'epoch'` differently than a scheduler stepped once per epoch.
